lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java

```diff
@@ -170,24 +175,43 @@ public Map<String, Object> handleRequest(Map<String, String> input, Context cont
                             apiKey,
                             credentialIssuerId);
 
+            auditEventUser =
+                    new AuditEventUser(
+                            userId,
+                            ipvSessionItem.getIpvSessionId(),
+                            clientOAuthSessionItem.getGovukSigninJourneyId(),
+                            ipAddress);
+
             if (VerifiableCredentialStatus.PENDING.equals(
                     verifiableCredentialResponse.getCredentialStatus())) {
                 processPendingResponse(
                         userId,
                         credentialIssuerId,
                         credentialIssuerConfig,
                         verifiableCredentialResponse,
-                        criOAuthState,
+                        criOAuthSessionItem.getCriOAuthSessionId(),
                         ipvSessionItem);
+                sendCriResRetrievedAuditEvent(
+                        auditEventUser,
+                        credentialIssuerId,
+                        CriResourceRetrievedType.PENDING.getType());
             } else {
+                List<SignedJWT> verifiableCredentials =
+                        verifiableCredentialResponse.getVerifiableCredentials();
                 processVerifiableCredentials(
                         userId,
                         credentialIssuerId,
                         credentialIssuerConfig,
                         ipAddress,
-                        verifiableCredentialResponse.getVerifiableCredentials(),
+                        verifiableCredentials,
                         clientOAuthSessionItem,
                         ipvSessionItem);
+                if (!verifiableCredentials.isEmpty()) {
+                    sendCriResRetrievedAuditEvent(
+                            auditEventUser,
+                            credentialIssuerId,
+                            CriResourceRetrievedType.VC.getType());
+                }
```

Comment:
This ordering means that we will log this event _after_ the VC received event - is that what we expect?

Reply:
yes currently this new event sent after IPV_VC_RECEIVED. Will check this.